src/vit-testing/mainnet-tools/src/snapshot_wormhole/command.rs

```diff
@@ -0,0 +1,199 @@
+use crate::snapshot_wormhole::Config;
+use color_eyre::eyre;
+use color_eyre::eyre::Result;
+use job_scheduler_ng::{Job, JobScheduler};
+use jormungandr_automation::jormungandr::LogLevel;
+use jortestkit::prelude::WaitBuilder;
+use snapshot_lib::RawSnapshot;
+use snapshot_trigger_service::client::SnapshotResult;
+use snapshot_trigger_service::{client::rest::SnapshotRestClient, config::JobParameters};
+use std::path::{Path, PathBuf};
+use std::str::FromStr;
+use std::time::{Duration, SystemTime, UNIX_EPOCH};
+use structopt::StructOpt;
+use thiserror::Error;
+use tracing::{debug, error, info, instrument, level_filters::LevelFilter};
+use tracing_subscriber::FmtSubscriber;
+use vit_servicing_station_lib::v0::endpoints::snapshot::RawSnapshotInput;
+use vit_servicing_station_tests::common::{
+    clients::RestClient, raw_snapshot::RawSnapshot as RawSnapshotRequest,
+};
+use voting_tools_rs::Output;
+
+/// Snapshot wormhole command which schedules snapshot file transport from snapshot trigger service to
+/// servicing station
+#[derive(StructOpt, Debug)]
+pub struct Command {
+    /// Path to configuration file
+    #[structopt(short, long)]
+    pub config: PathBuf,
+
+    /// Log level
+    #[structopt(long = "log-level", default_value = "INFO")]
+    pub log_level: LogLevel,
+
+    #[structopt(subcommand)]
+    cmd: Operation,
+}
+
+/// Sub command to run. Either 'one-shot' job, which ends program after single job is done, or 'schedule'
+/// which will run job continuously based on cron string
+///
+/// WARNING: there is custom cron string used which allows to program scheduler based on seconds.
+/// The scheduling format is as follows:
+///
+/// sec   min   hour   day of month   month   day of week   year
+/// *     *     *      *              *       *             *
+#[derive(StructOpt, Debug)]
+pub enum Operation {
+    OneShot,
+    Schedule(Schedule),
+}
+impl Command {
+    /// Executes command
+    ///
+    /// # Errors
+    ///
+    /// On IO related errors
+    pub fn exec(self) -> Result<()> {
+        color_eyre::install()?;
```

Comment:
This should probably be in `main()`, since it can only be called once globally. Registering a panic hook is something that library code probably shouldn't do IMO